server/partition.go

```diff
@@ -91,6 +108,9 @@ type partition struct {
 	autoPauseTime                 time.Duration
 	autoPauseDisableIfSubscribers bool
 	subscriberCount               int64
+	messagesTimestamps            eventTimestamps // First and latest time a message was received on this partition
```

Comment:
I suggest rename it to `messagesProcessTimestamp` or something equivalent. Reason is that in general with streaming system, we may have to work with 3 typical timestamps:

- Event timestamp : It's emission timestamp, normally should be stamped on client publisher
- Received timestamp: It records the moment the event is recorded on the server
- Process timestamp: It records the moment the event is processed.

In this case I guess process and received would be the same, as the partition gets the message and stores it right away.

Minor point but I guess it's clearer

Reply:
I see your point. I have renamed this in 3411c72.